tools/generate_package_config/bin/generate_from_legacy.dart

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+// @dart=2.8
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:package_config/package_config.dart';
+import 'package:package_config/packages_file.dart'; // ignore: deprecated_member_use
+import 'package:package_config/src/package_config_json.dart';
+import 'package:pub_semver/pub_semver.dart';
+import 'package:yaml/yaml.dart';
+
+void main(List<String> args) async {
+  if (args.length != 1) {
+    throw ArgumentError('Unexpected arguments $args\n\n$usage');
+  }
+  if (args.first == 'help') {
+    print(usage);
+    return;
+  }
+  var packagesUri = Uri.base.resolve(args.first);
+  var packagesFile = File.fromUri(packagesUri);
+  if (!await packagesFile.exists()) {
+    throw ArgumentError('Unable to read file at `$packagesUri`');
+  }
+  var packageMap = parse(await packagesFile.readAsBytes(), packagesFile.uri);
+  var packages = <Package>[];
+  for (var packageEntry in packageMap.entries) {
+    var name = packageEntry.key;
+    var uri = packageEntry.value;
+    if (uri.scheme != 'file') {
+      throw ArgumentError(
+          'Only file: schemes are supported, but the `$name` package has '
+          'the following uri: ${uri}');
+    }
+    Uri packageRoot;
+    Uri pubspec;
+    LanguageVersion languageVersion;
+    // Pub packages will always have a path ending in `lib/`, and a pubspec
+    // directly above that.
+    if (uri.path.endsWith('lib/')) {
+      packageRoot = uri.resolve('../');
+      pubspec = packageRoot.resolve('pubspec.yaml');
+      languageVersion = await languageVersionFromPubspec(pubspec, name);
+    }
+    if (languageVersion == null) {
```

Comment:
Ignoring packages without a pubspec.yaml

Reply:
I think we probably do want to create an entry for them still though so that they can be resolved? They will get the latest language version which is the specified behavior if they have no explicit one.